File: lib/core/constants/canonical_allergens.dart
// lib/core/constants/canonical_allergens.dart
//
// Fixed allergen vocabulary used across the OCR + Gemini extraction pipeline
// (core/utils, data/services/product_extraction_service.dart) and mirrored
// in the admin dashboard's review checklist. Extraction is instructed to
// match detected allergen statements against this list rather than
// returning free-text, so "Milk" / "milk" / "contains milk" / "dairy" all
// collapse to one consistent value instead of admin (and the app) having to
// deal with N different spellings of the same allergen.
//
// Keep this list and the admin dashboard's copy of it in sync -- see
// admin-claro/src/constants/canonicalAllergens.js.
class CanonicalAllergens {
  CanonicalAllergens._();

  static const List<String> values = [
    'Milk',
    'Eggs',
    'Fish',
    'Shellfish',
    'Tree Nuts',
    'Peanuts',
    'Wheat',
    'Soy',
    'Sesame',
  ];
}

File: src/constants/canonicalAllergens.js
// src/constants/canonicalAllergens.js
//
// Fixed allergen vocabulary used by the admin review checklist. Must stay
// in sync with the OCR + Gemini extraction pipeline's allergen list --
// mirrors lib/core/constants/canonical_allergens.dart (Flutter app) and
// tools/bulk-import/import.js's CANONICAL_ALLERGENS constant. Extraction
// only ever returns values from this list, so the checklist here always
// has something to pre-check against -- see PROMPT_SCHEMA.md in
// bulk-import for the full shared-contract note.

export const CANONICAL_ALLERGENS = [
  "Milk",
  "Eggs",
  "Fish",
  "Shellfish",
  "Tree Nuts",
  "Peanuts",
  "Wheat",
  "Soy",
  "Sesame",
];
